### ISSUE TITLE
The errorMessage will not be cleared after the controlled Form formData changes

### ISSUE THREAD

**Issue by JinYuSha0** [rjsf-team/react-jsonschema-form#4426]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

core

### Version

5.23.2

### Current Behavior

![rjsf-issuse](https://github.com/user-attachments/assets/1d1ac905-2f30-482e-96bf-47b468f91b58)

[Codesandbox](https://codesandbox.io/p/github/JinYuSha0/rjsf-issuse/main)

### Expected Behavior

clear error message after onChange, similar to the case where formData is not used
![rjsf-issuse2](https://github.com/user-attachments/assets/00d00a4a-0ea1-4e9b-9d33-663654ba94d0)

### Environment

```markdown
- OS:Windows 22H2
- Node:v22.11.0
- npm:10.8.3
```

**Comment by abdalla-rko** [comment]
When you don't have the `**liveValidate**` option enabled the rjsf will only validate on submit. If you want to have live validation, please enable the `**liveValidate**` option (see [demo ](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwicmVxdWlyZWQiOlsicmVxdWlyZWRGaWVsZCJdLCJwcm9wZXJ0aWVzIjp7InJlcXVpcmVkRmllbGQiOnsidHlwZSI6InN0cmluZyJ9fX0sInVpU2NoZW1hIjp7fSwidGhlbWUiOiJhbnRkIiwibGl2ZVNldHRpbmdzIjp7InNob3dFcnJvckxpc3QiOiJ0b3AiLCJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGVkIjpmYWxzZSwibm9IdG1sNVZhbGlkYXRlIjpmYWxzZSwicmVhZG9ubHkiOmZhbHNlLCJvbWl0RXh0cmFEYXRhIjpmYWxzZSwibGl2ZU9taXQiOmZhbHNlLCJleHBlcmltZW50YWxfZGVmYXVsdEZvcm1TdGF0ZUJlaGF2aW9yIjp7ImFycmF5TWluSXRlbXMiOiJwb3B1bGF0ZSIsImFsbE9mIjoic2tpcERlZmF1bHRzIiwiY29uc3RBc0RlZmF1bHRzIjoiYWx3YXlzIiwiZW1wdHlPYmplY3RGaWVsZHMiOiJwb3B1bGF0ZUFsbERlZmF1bHRzIiwibWVyZ2VEZWZhdWx0c0ludG9Gb3JtRGF0YSI6InVzZUZvcm1EYXRhSWZQcmVzZW50In0sImxpdmVWYWxpZGF0ZSI6dHJ1ZX19) with live validate)

**Comment by JinYuSha0** [comment]
Yes I know, If I enable **liveValidate**,  errorMessage will be display when i open the page, I think it would be more reasonable to display it after interaction.
![rjsf-issuse3](https://github.com/user-attachments/assets/3c353858-4a41-4172-8b38-b98b4d240357)
